doc/compiler/optimizer/IntroReadLogFile.md

```diff
@@ -0,0 +1,1188 @@
+ n1858n   (  2)    ==>l2a (in &GPR_0562) (X!=0 sharedMemory )
+ n1861n   (  0)    aconst NULL (X==0 sharedMemory )                                                   [0x7fef25c7d570] bci=[9,1,-] rc=0 vc=4181 vn=- li=479 udi=- nc=0 flg=0x2
+ n4935n   (  1)    GlRegDeps ()                                                                       [0x7fef26429640] bci=[9,1,-] rc=1 vc=4181 vn=- li=479 udi=- nc=6 flg=0x20
+ n4929n   (  1)      ==>iRegLoad (in GPR_0512) (SeenRealReference )
+ n4930n   (  3)      ==>iRegLoad (in GPR_0528) (cannotOverflow SeenRealReference )
+ n4931n   (  1)      ==>aRegLoad (in &GPR_0544) (SeenRealReference sharedMemory )
+ n4932n   (  1)      ==>iRegLoad (in GPR_0545) (SeenRealReference )
+ n4933n   (  1)      ==>aRegLoad (in &GPR_0560) (X!=0 X>=0 SeenRealReference sharedMemory )
+ n4934n   (  0)      ==>aRegLoad (in &GPR_0561)
+------------------------------
+
+ [0x7fef12b0abc0]	test	&GPR_0562, &GPR_0562		# TEST8RegReg
+ [0x7fef12b0b080]	assocreg                        # assocreg
+	POST: [&GPR_0544 : ebx] [GPR_0512 : edi] [GPR_0528 : r8d] [GPR_0545 : r9d] [&GPR_0560 : r12d] [&GPR_0561 : r13d]
+ [0x7fef12b0ae10]	je	Label L0025			# JE4
+	 PRE: [&GPR_0561 : r13d] [&GPR_0560 : r12d] [GPR_0545 : r9d] [&GPR_0544 : ebx] [GPR_0528 : r8d] [GPR_0512 : edi]
+	POST: [&GPR_0561 : r13d] [&GPR_0560 : r12d] [GPR_0545 : r9d] [&GPR_0544 : ebx] [GPR_0528 : r8d] [GPR_0512 : edi]
+
+…
+============================================================
+; Live regs: GPR=5 FPR=0 VRF=0 {&GPR_0562, GPR_0545, &GPR_0544, GPR_0528, GPR_0512}
+------------------------------
+ n4736n   (  0)  astore  <temp slot 8>[#522  Auto] [flags 0x4007 0x0 ]                                [0x7fef26425810] bci=[9,1,-] rc=0 vc=4181 vn=- li=356 udi=- nc=1
+ n1858n   (  2)    ==>l2a (in &GPR_0562) (X!=0 sharedMemory )
+------------------------------
+------------------------------
+ n4736n   (  0)  astore  <temp slot 8>[#522  Auto] [flags 0x4007 0x0 ]                                [0x7fef26425810] bci=[9,1,-] rc=0 vc=4181 vn=- li=356 udi=- nc=1
+ n1858n   (  1)    ==>l2a (in &GPR_0562) (X!=0 sharedMemory )
+------------------------------
+
+ [0x7fef12b0bdd0]	mov	qword ptr [vfp], &GPR_0562		# S8MemReg, SymRef  <temp slot 8>[#678  Auto] [flags 0x4007 0x0 ]
+…
+```
+
+`n1858n l2a` is evaluated and the value is stored in `&GPR_0562` in the instruction at `0x7fef12b0a7c0`.
+`&` means the virtual register is a collected reference. The reference needs to be conveyed in a garbage collector.
+
+`GPR_0562` is used directly for the `NULL` test in `n1862n ifacmpeq` in the instruction at `0x7fef12b0abc0`.
+
+`GPR_0562` is used again in `n4736n astore` to store the value to stack temp slot `8` in the instruction at
+`0x7fef12b0bdd0`. `vfp` stands for virtual frame pointer. It is used because we have not mapped stack slots.
+We don’t have offset associated with each auto slot yet.
+
+### 7.3 Post Instruction Selection
+
+```
```

Comment:
I think this section could use a brief summary paragraph describing the steps listed in the examples below, just to set the stage, and explain what is happening.

Reply:
I realized the content in 7.3 are examples of the transformations after instruction section. I'll change 7.3 as below: 

> ### 7.3 Examples Of Sections After Instruction Selection
> 
> The following are some examples of transformations that happen after Instruction Selection.
> For example for the instruction at `0x7fef12b0bdd0` from the above example:

Reply:
Addressed in 9c5fd3b